类型检查+Lint+单元测试失败: PR #733 引入 Zod errorMap 和 curly 规则问题

## CI 监控发现测试失败

**触发提交:** 
- b49c738 Merge pull request #733 from hs3180/fix/issue-729-send-user-feedback-validation
- 57ee520 fix(mcp): improve send_user_feedback parameter validation error messages (Issue #729)

**失败文件:** `src/mcp/feishu-context-mcp.ts`

---

## 1. 类型检查失败

```
src/mcp/feishu-context-mcp.ts(1109,17): error TS2769: No overload matches this call.
  Overload 1 of 2, '(values: readonly [\"text\", \"card\"], params?: string | { error?: string | $ZodErrorMap<$ZodIssueInvalidValue<unknown>> | undefined; message?: string | undefined; } | undefined): ZodEnum<...>', gave the following error.
    Object literal may only specify known properties, and 'errorMap' does not exist in type '{ error?: string | $ZodErrorMap<$ZodIssueInvalidValue<unknown>> | undefined; message?: string | undefined; }'.
```

**问题代码 (line 1109-1111):**
```typescript
format: z.enum(['text', 'card'], {
  errorMap: () => ({ message: 'format is REQUIRED. Use "text" for plain text messages or "card" for interactive cards.' }),
}).describe('REQUIRED: "text" for plain text, "card" for interactive cards. This parameter is mandatory.'),
```

**原因:** Zod enum 的第二个参数不支持 `errorMap` 属性，应使用 `message` 或其他方式。

---

## 2. Lint 检查失败

```
/Users/hs3180/clawd/disclaude/src/mcp/feishu-context-mcp.ts
  1127:33  error  Expected { after 'if' condition  curly
  1128:33  error  Expected { after 'if' condition  curly
  1129:33  error  Expected { after 'if' condition  curly
```

**问题代码 (lines 1127-1129):**
```typescript
if (!('config' in obj)) missing.push('config');
if (!('header' in obj)) missing.push('header');
if (!('elements' in obj)) missing.push('elements');
```

**原因:** ESLint `curly` 规则要求 `if` 语句后必须使用花括号 `{}`。

---

## 3. 单元测试失败

```
 FAIL  src/mcp/feishu-mcp-server.test.ts > Feishu MCP Server > Tool Definitions > should define send_user_feedback tool with correct schema
AssertionError: expected 'Send a message to a Feishu chat...' to contain 'Card Format Requirements'
```

**原因:** 测试期望 description 包含 `'Card Format Requirements'`，但新代码使用的是 `'Card Structure Requirements'`。

**测试代码 (line 83):**
```typescript
expect(sendFeedbackTool?.description).toContain('Card Format Requirements');
```

---

## 建议修复

1. **类型检查:** 将 `errorMap` 改为 `message` 参数，或使用 `.refine()` 进行自定义验证
2. **Lint:** 为单行 `if` 语句添加花括号
3. **测试:** 更新测试中的期望字符串，或更新代码中的标题以匹配测试

🤖 由 CI Monitor 自动检测

--- a/src/mcp/feishu-context-mcp.ts
+++ b/src/mcp/feishu-context-mcp.ts
@@ -1069,7 +1069,7 @@ export const feishuToolDefinitions: InlineToolDefinition[] = [
 
 ---
 
-## Card Structure Requirements
+## Card Format Requirements
 
 When \`format: "card"\`, content MUST include:
 - \`config\`: Object (e.g., \`{"wide_screen_mode": true}\`)
@@ -1107,7 +1107,7 @@ When parentMessageId is provided, the message is sent as a reply to that message
     parameters: z.object({
       content: z.union([z.string(), z.object({}).passthrough()]).describe('The content to send. MUST match format type: string for "text", object for "card" with {config, header, elements}.'),
       format: z.enum(['text', 'card'], {
-        errorMap: () => ({ message: 'format is REQUIRED. Use "text" for plain text messages or "card" for interactive cards.' }),
+        message: 'format is REQUIRED. Use "text" for plain text messages or "card" for interactive cards.',
       }).describe('REQUIRED: "text" for plain text, "card" for interactive cards. This parameter is mandatory.'),
       chatId: z.string().describe('Feishu chat ID (get this from the task context/metadata)'),
       parentMessageId: z.string().optional().describe('Optional parent message ID for thread replies.'),
@@ -1124,9 +1124,9 @@ When parentMessageId is provided, the message is sent as a reply to that message
       if (format === 'card' && typeof content === 'object' && content !== null) {
         const obj = content as Record<string, unknown>;
         const missing: string[] = [];
-        if (!('config' in obj)) missing.push('config');
-        if (!('header' in obj)) missing.push('header');
-        if (!('elements' in obj)) missing.push('elements');
+        if (!('config' in obj)) { missing.push('config'); }
+        if (!('header' in obj)) { missing.push('header'); }
+        if (!('elements' in obj)) { missing.push('elements'); }
         if (missing.length > 0) {
           return toolSuccess(`❌ Card validation failed: missing required fields: ${missing.join(', ')}.\n\nRequired structure:\n{"config": {...}, "header": {"title": {...}, ...}, "elements": [...]}`);
         }

